dist/nix/vimrc-with-plugins.nix

```diff
@@ -0,0 +1,28 @@
+{ pkgs ? import <nixpkgs>{}
+, additionalPlugins? []
+, additionalCustPlugins? {}
+, additionalVimrc? ""#(pkgs.builtins.readFile ./vimrc)
+}:
+with pkgs;
+let quick-lint-js = import ./quick-lint-js.nix { inherit pkgs;} ;
+in
+vimUtils.vimrcFile {
+  customRC =
+    ''
+      let g:ale_javascript_quick_lint_js_executable = '${quick-lint-js}/bin/quick-lint-js'
+    ''+additionalVimrc;
+   vam.knownPlugins = vimPlugins // additionalCustPlugins // {
+     quick-lint-js = vimUtils.buildVimPluginFrom2Nix {
+       name = "quick-lint-js";
+       src = "${quick-lint-js}/share/vim/vimfiles";
+     };
```

Comment:
Could we have this part express the dependency on ALE? Maybe something like this (based on `vim-addon-xdebug` in `<nixpkgs>/pkgs/misc/vim-plugins/overrides.nix`):

    dependencies = with self; [ ale ];